fvm/src/kernel/default.rs

```diff
@@ -1006,113 +1006,93 @@ impl<C> EventOps for DefaultKernel<C>
 where
     C: CallManager,
 {
-    fn emit_event(&mut self, raw_evt: &[u8], raw_key: &[u8], raw_val: &[u8]) -> Result<()> {
-        const MAX_NR_ENTRIES: usize = 256;
-        const MAX_KEY_LEN: usize = 32;
+    fn emit_event(
+        &mut self,
+        event_headers: &[fvm_shared::sys::EventEntry],
```

Comment:
Taking this as a slice of event entries to avoid unsafe logic in the kernel where possible.